src/plugins/intel_npu/tests/functional/behavior/work_with_devices.hpp

```diff
@@ -73,6 +73,36 @@ TEST_P(TestCompiledModelNPU, samePlatformProduceTheSameBlob) {
     }
 }
 
+TEST_P(TestCompiledModelNPU, samePlatformProduceTheSameBlobCacheEnabled) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED() {
+        std::string platform = ov::test::utils::getTestsPlatformFromEnvironmentOr("3720");
+
+        // compiled model (uncached)
+        configuration[ov::intel_npu::defer_weights_load.name()] = true;
+        auto configuration1 = configuration;
+        configuration1[ov::intel_npu::platform.name()] = platform;
+        const auto& ov_model1 = buildSingleLayerSoftMaxNetwork();
+        auto compiled_model1 = core->compile_model(ov_model1, target_device, configuration1);
+        std::stringstream blobStream1;
+        compiled_model1.export_model(blobStream1);
+
+        // cached blobs
+        auto configuration2 = configuration;
+        configuration2[ov::intel_npu::platform.name()] = platform;
+        configuration2[ov::intel_npu::bypass_umd_caching.name()] = false;
+        const auto& ov_model2 = buildSingleLayerSoftMaxNetwork();
+        auto compiled_model2 = core->compile_model(ov_model2, target_device, configuration2);
+        auto compiled_model3 = core->compile_model(ov_model2, target_device, configuration2);
+        std::stringstream blobStream2, blobStream3;
+        compiled_model2.export_model(blobStream2);
+        compiled_model3.export_model(blobStream3);
+
+        ASSERT_NE(0, blobStream1.str().size());
+        ASSERT_EQ(0, std::memcmp(blobStream1.str().c_str(), blobStream2.str().c_str(), blobStream1.str().size()));
+        ASSERT_EQ(0, std::memcmp(blobStream2.str().c_str(), blobStream3.str().c_str(), blobStream2.str().size()));
```

Comment:
```suggestion
        ASSERT_EQ(0, std::memcmp(blobStream1.str().c_str(), blobStream3.str().c_str(), blobStream1.str().size()));
```

Is this enough for testing compiled model 1 from the compiler vs compiled model 3 from UMD caching?